packages/server-wallet/src/wallet/multi-threaded-wallet/worker.ts

```diff
@@ -22,45 +22,46 @@ const walletConfig: ServerWalletConfig = {
 const logger = createLogger(walletConfig);
 
 logger.debug(`Worker %o starting`, threadId);
-const wallet = Wallet.create(walletConfig);
-parentPort?.on('message', async (message: any) => {
-  if (isMainThread) {
-    parentPort?.postMessage(
-      left(new Error('Attempting to execute worker thread script on the main thread'))
-    );
-  }
+Wallet.create(walletConfig).then(wallet => {
```

Comment:
Another option is to factor out the bulk of this file into an `async` function and then call the function (without awaiting).

Reply:
Done in 60b5400